h/test/paginator_test.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+
+from pyramid.testing import DummyRequest
+import pytest
+
+from h.paginator import paginate
+
+
+class FakeQuery(object):
+
+    """
+    A helper class to fake out the fluent API of a SQLAlchemy Query object.
+    """
+
+    def __init__(self, total):
+        self.total = total
+        self.offset_param = None
+        self.limit_param = None
+        self.all_called = False
+
+    def count(self):
+        return self.total
+
+    def offset(self, n):
+        self.offset_param = n
+        return self
+
+    def limit(self, n):
+        self.limit_param = n
+        return self
+
+    def all(self):
+        self.all_called = True
+        return self
+
+
+def fake_view(total):
+    """
+    Make a fake view function and a mock query object for testing.
+    """
+    query = FakeQuery(total)
+
+    def view(context, request):
+        return query
+
+    return query, view
+
+
+def test_paginate_returns_resultset():
+    request = DummyRequest()
+    query, view = fake_view(total=10)
+
+    wrapped = paginate(view)
+    result = wrapped(context=None, request=request)
+
+    assert result['results'] == query
+    # Assert that .all() has been called to turn the query into a list.
+    assert query.all_called
```

Comment:
Minor quibble. I generally prefer to verify only what ultimately gets returned and any side-effects and not have the tests dependent on exactly how the data gets fetched.